bin/build-plugin-zip.sh

```diff
@@ -25,34 +35,50 @@ elif ! git diff --cached --exit-code > /dev/null; then
 fi
 if [ ! -z "$changed" ]; then
 	git status
-	echo "ERROR: Cannot build plugin zip with dirty working tree."
-	echo "       Commit your changes and try again."
-	exit 1
+	error "ERROR: Cannot build plugin zip with dirty working tree. ☝️
+       Commit your changes and try again."
+	exit
 fi
 
 branch="$(git rev-parse --abbrev-ref HEAD)"
 if [ "$branch" != 'master' ]; then
-	echo "WARNING: You should probably be running this script against the"
-	echo "         'master' branch (current: '$branch')"
-	echo
+	warning "WARNING: You should probably be running this script against the
+         'master' branch (current: '$branch')"
 	sleep 2
 fi
 
-# Remove ignored files to reset repository to pristine condition. Previous test
-# ensures that changed files abort the plugin build.
-status "Cleaning working directory..."
-git clean -xdf
+# Do a dry run of the repository reset. Prompting the user for a list of all
+# files that will be removed should prevent them from losing important files!
+status "Resetting the repository to pristine condition."
+echo "Would remove gutenberg.zip"
```

Comment:
Should it read "Would remove the following:"?

Reply:
It's in there to "blend in" with `git clean -xdf`'s output. So it'll look like:

<img width="1284" alt="screenshot 2018-06-28 14 11 23" src="https://user-images.githubusercontent.com/90871/42099040-5f3ff286-7bb4-11e8-8e17-76ae68802246.png">

But it's weird out of context; I'll add a code comment.